src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -8,29 +8,95 @@
 import java.net.URI;
 import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 
 public final class LogFileGroup {
-    @Getter
-    private List<LogFile> logFiles;
+    private final List<LogFile> logFiles;
     private static Map<String, LogFile> fileHashToLogFile;
     @Getter
     private final Pattern filePattern;
     private final URI directoryURI;
     private final Instant lastUpdated;
 
-    private LogFileGroup(List<LogFile> files, Pattern filePattern, URI directoryURI, Instant lastUpdated) {
-        this.logFiles = files;
+    /**
+     * Creates a log file group without initializing the logFiles. To initialize them you must call
+     * getLogFiles.
+     * @param logFiles - the log group log files
+     * @param filePattern - the pattern use to match files on the provided directoryURI
+     * @param directoryURI - the directory where log files are located
+     * @param lastUpdated - instant used to filter out log files last modified after its value
+     */
+    public LogFileGroup(List<LogFile> logFiles, Pattern filePattern, URI directoryURI, Instant lastUpdated) {
+        this.logFiles = logFiles;
```

Comment:
Not understanding this. I had thought the `LogFileGroup` would create the hardlinks which back the `List<LogFile>`?

Reply:
Yes I mentioned on the description that will be the next PR